Implemented Enchainment.

joncloud opened the pull request.

teschnei commented:
eh.. not really how it should be done. you don't really want people popping enchainment to get tp bonus on all their WS right?
i would check to see where blue magic gets their TP from (hopefully its in one function...) and check for enchainment there

joncloud commented:
I updated the core blue magic calculation.  It looks like Chain Affinity also did not adjust based on the current caster's TP.  I implemented it using BluefTP like the TODO comment suggested.

teschnei commented on scripts/globals/bluemagic.lua:
```diff
@@ -101,9 +101,19 @@ function BluePhysicalSpell(caster, target, spell, params)
+    local chainAffinity = caster:getStatusEffect(EFFECT_CHAIN_AFFINITY);
+    if chainAffinity ~= nil then
+        -- Calculate the total TP available for the fTP multiplier.
+        local tp = caster:getTP() + caster:getMerit(MERIT_ENCHAINMENT);
```
just need to make sure tp doesn't go over the maximum and then it's good

joncloud commented:
I did not catch anywhere that declared a constant, so I put a literal in for capping.
